workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/TransformSendSystem.cs

```diff
@@ -1,59 +1,78 @@
+    public class TransformSendSystem : CustomSpatialOSSendSystem<TransformInternal.Component>
     {
         private struct TransformData
         {
             public readonly int Length;
-            public ComponentDataArray<Transform.Component> Transforms;
-            [ReadOnly] public ComponentDataArray<Authoritative<Transform.Component>> TransformAuthority;
+            public ComponentDataArray<TransformInternal.Component> Transform;
+            public ComponentDataArray<LastTransformSentData> LastTransformSent;
+            [ReadOnly] public ComponentDataArray<Authoritative<TransformInternal.Component>> TransformAuthority;
             [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
         }
 
         [Inject] private TransformData transformData;
 
-        // Number of transform sends per second.
-        private const float SendRateHz = 30.0f;
-
-        private float timeSinceLastSend = 0.0f;
-
         protected override void OnUpdate()
         {
-            // Send update at SendRateHz.
-            timeSinceLastSend += Time.deltaTime;
-            if (timeSinceLastSend < 1.0f / SendRateHz)
+            for (var i = 0; i < transformData.Length; i++)
             {
-                return;
-            }
+                var transform = transformData.Transform[i];
 
-            timeSinceLastSend = 0.0f;
+                if (transform.DirtyBit != true)
+                {
+                    continue;
+                }
 
-            for (var i = 0; i < transformData.Length; i++)
-            {
-                var component = transformData.Transforms[i];
+                var lastTransformSent = transformData.LastTransformSent[i];
+                lastTransformSent.TimeSinceLastUpdate += Time.deltaTime;
+                transformData.LastTransformSent[i] = lastTransformSent;
 
-                if (component.DirtyBit != true)
+                if (lastTransformSent.TimeSinceLastUpdate <
+                    1.0f / TransformSynchronizationConfig.MaxTransformUpdateRateHz)
                 {
                     continue;
                 }
 
+                // Need to be doing things with velocity and orientation too for this to work
+                // var squareDisatnce =
+                //     TransformUtils.SquareDisatnce(lastTransformSent.Transform.Location,
+                //         transformData.Transform[i].Location);
+                //
+                // if (squareDisatnce == 0.0f)
+                // {
+                //     continue;
+                // }
+                //
+                // if (lastTransformSent.TimeSinceLastUpdate <
+                //     TransformSynchronizationConfig.MaxTimeForStalePositionWithoutUpdateS)
+                // {
+                //     if (squareDisatnce < TransformSynchronizationConfig.MaxSquarePositionChangeWithoutUpdate)
+                //     {
+                //         continue;
+                //     }
+                // }
```

Comment:
Commented code

Reply:
that is there as an example. Also there is a lot of work that needs to be done on this. We can remove it but then it all has to be in a branch on my machine.